### DISCUSSION TITLE
Does (or will) this support multiple users, permissions, and collaborative editing?

### DISCUSSION THREAD

**Discussion by looeee** [toeverything/AFFiNE#97] [General]
I've tested the [Windows build here](https://github.com/m1911star/affine-client/releases) and this looks like a really cool app. 

That build automatically logs you in as a single user which is fine for testing. But it doesn't allow me to test some of the most important features of Notion which allow it to be used for a team. That would require multiple users (preferably using OIDC to login), setting permissions on documents, and optionally but nice to have, collaborative editing. 

If I install this on a VPS and make it available at `affine.mysite.com`, for example, will I able to use it for a team/business? Are these features available or planned?

**Comment by HeJiachen-PM** [comment]
Hi looeee，
> I've tested the [Windows build here](https://github.com/m1911star/affine-client/releases) and this looks like a really cool app.

Thanks! 


> That build automatically logs you in as a single user which is fine for testing. But it doesn't allow me to test some of the most important features of Notion which allow it to be used for a team. That would require multiple users (preferably using OIDC to login), setting permissions on documents, and optionally but nice to have, collaborative editing.

The build is NOT developed by us and should be considered as a live demo not a production-ready build of Affine. We will be releasing a web version soon. Yes, we will support collaborative editing for teams and it will be available once our first beta version is out. We are also preparing a roadmap site for details on this so please bear with us before it is out.


> If I install this on a VPS and make it available at affine.mysite.com, for example, will I able to use it for a team/business? Are these features available or planned?

That really depends on how you use it for business. We are fully open-sourced and currently under MIT license so you are welcome to self-host Affine or modify it the way you want, and we will keep on making self-hosting free and easier to use. However, the free version will not support storage on our cloud or syncing with multiple users through our cloud. Cloud-based analysis and collaboration will be in our commercial plan for a better experience.
However, that does not mean that collaboration features are completely commercial since we will be supporting end-2-end connection through CRDTs.

**Reply by seankhl** [reply]
> [...] we will be supporting end-2-end connection through CRDTs.

I notice you didn't say "encryption". Is end-to-end encryption (e2ee) on the roadmap? I feel as a "privacy-focused" app, e2ee should be available.

Supporting multi-device collaboration (same user on multiple devices as well as multiple users) + e2ee, even if it's not fully real-time but as long as the updates are consistent, would be a uniquely-differentiating feature that, IMO, would drive a lot of users toward you. Please seriously consider it. If it can be a confimed future feature, I would seriously consider making significant contributions.

**Reply by HeJiachen-PM** [reply]
Yes, end-to-end encryption is indeed planned with high priority. However, it may not be open-sourced and may not be free. We have supported consistent semantic merge and will be supporting near-real-time collaboration with no conflict to local-storage soon, which will all be free but features that are cloud-related are likely to be non-free.

**Reply by looeee** [reply]
@HeJiachen-PM do you have a document outlining what will be free and what will be part of the paid license? Likewise what features won't be available when self hosting. Those are things I would need to know before jumping on board. 

Also, will it be possible to use the paid license when self hosting the app? I'm not against paying you if the price is reasonable :) 
However I won't be interested in a hosted version of the app.

**Reply by looeee** [reply]
> That really depends on how you use it for business. 

I want to use it as a self hosted Notion replacement so the more of the Notion's features that are support the better really. For context I currently use getoutline.com and from a user perspective, it's close enough to a drop in replacement for Notion that it's been easy to transition for my team. In the self hosted version, user accounts, permissions, and collaborative editing are all supported out of the box. 

The reason I am still searching for an alternative is that it's not an easy app to self host, and many questions related to self hosting on the Outline GH get waived away with "it works fine on our hosted app" or "just implement it yourself" - this includes many questions related to it being difficult to install but also bugs like the import/export not working on Windows computers. If you search around for information on self hosting Outline, you will find the general consensus from devs is that it's great once you have set it up but an absolute _pain_ to install and administer or tweak. 

> We are fully open-sourced and currently under MIT license so you are welcome to self-host Affine or modify it the way you want

I do understand that but unfortunately I have limited time. Are you saying that multiple users and permissions on the self hosted app is something I would have to implement myself?

**Comment by m1911star** [comment]
> I've tested the [Windows build here](https://github.com/m1911star/affine-client/releases) and this looks like a really cool app.

Glad you like it！
